src/map/clif.cpp

```diff
@@ -4671,17 +4671,14 @@ void clif_tradeadditem( struct map_session_data* sd, struct map_session_data* ts
 ///     0 = success
 ///     1 = overweight
 ///     2 = trade canceled
-void clif_tradeitemok(struct map_session_data* sd, int index, int fail)
+void clif_tradeitemok(struct map_session_data& sd, int index, e_exitem_add_result result)
 {
-	int fd;
-	nullpo_retv(sd);
+	PACKET_ZC_ACK_ADD_EXCHANGE_ITEM* p = reinterpret_cast<PACKET_ZC_ACK_ADD_EXCHANGE_ITEM*>(packet_buffer);
+	p->packetType = HEADER_ZC_ACK_ADD_EXCHANGE_ITEM;
+	p->index = client_index(index);
+	p->result = static_cast<uint8>(result);
 
-	fd = sd->fd;
-	WFIFOHEAD(fd,packet_len(0xea));
-	WFIFOW(fd,0) = 0xea;
-	WFIFOW(fd,2) = index;
-	WFIFOB(fd,4) = fail;
-	WFIFOSET(fd,packet_len(0xea));
+	clif_send(p, sizeof(p), &sd.bl, SELF);
```

Comment:
```suggestion
	clif_send(&p, sizeof(p), &sd.bl, SELF);
```
sizeof(p) was wrong in the pointer variant. sizeof(*p) would have been correct. Therefore the rest can stay.